### PR TITLE
chore: add rudder-cli source and rm binaries

### PR DESCRIPTION
**Pull request by lvrach**
# Description

* Adding [rudderlabs/rudder-cli](https://github.com/rudderlabs/rudder-cli) repository to `rudder-server/cmd`
* Remove binary images
* Change Dockerfile to build rudder-cli instead of coping binaries 

## Linear Ticket

https://linear.app/rudderstack/issue/PIPE-256/move-rudder-cli-to-rudder-server

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by achettyiitr — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -103,6 +107,11 @@ require (
 	google.golang.org/protobuf v1.31.0
 )
 
+require (
+	github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e // indirect
+	github.com/kataras/tablewriter v0.0.0-20180708051242-e063d29b7c23 // indirect
+)
```
Comment (atzoum) [inline go.mod:113]: please move these to the indirect group below

File: go.mod [hunk 2]
```diff
@@ -84,6 +87,7 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/tidwall/gjson v1.16.0
 	github.com/tidwall/sjson v1.2.5
+	github.com/urfave/cli v1.22.2
```
Comment (atzoum) [inline go.mod:90]: can we skip v1 and just use v2?

Comment (lvrach) [reply go.mod:90]: I will create a task to address this.

Comment (lvrach) [reply go.mod:90]: v1 is seems to be supported btw https://github.com/urfave/cli/releases/tag/v1.22.14, so it not an urgent security reason to upgrade.

**Review by atzoum — APPROVED** [review]

**Pull request merged**